File: etamen-backend/app/Modules/Radiology/Infrastructure/Models/RadiologyScan.php
<?php

namespace App\Modules\Radiology\Infrastructure\Models;

use App\Models\User;
use App\Modules\Providers\Domain\Enums\ProviderStatus;
use App\Modules\Providers\Domain\Enums\ProviderType;
use App\Modules\Providers\Infrastructure\Models\Provider;
use App\Modules\Providers\Infrastructure\Models\ProviderBranch;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Validation\ValidationException;

class RadiologyScan extends Model
{
    protected $fillable = [
        'provider_id',
        'branch_id',
        'radiology_scan_category_id',
        'name_ar',
        'name_en',
        'description_ar',
        'description_en',
        'preparation_ar',
        'preparation_en',
        'duration_minutes',
        'base_price',
        'requires_preparation',
        'requires_fasting',
        'contrast_required',
        'home_available',
        'branch_available',
        'report_delivery_enabled',
        'is_active',
        'created_by',
        'updated_by',
    ];

    protected function casts(): array
    {
        return [
            'duration_minutes' => 'integer',
            'base_price' => 'decimal:2',
            'requires_preparation' => 'boolean',
            'requires_fasting' => 'boolean',
            'contrast_required' => 'boolean',
            'home_available' => 'boolean',
            'branch_available' => 'boolean',
            'report_delivery_enabled' => 'boolean',
            'is_active' => 'boolean',
        ];
    }

    protected static function booted(): void
    {
        static::saving(function (RadiologyScan $scan): void {
            $provider = Provider::query()->find($scan->provider_id);

            if (! $provider || $provider->type !== ProviderType::Radiology) {
                throw ValidationException::withMessages([
                    'provider_id' => ['The selected provider must be a radiology provider.'],
                ]);
            }

            if ($provider->status === ProviderStatus::Suspended) {
                throw ValidationException::withMessages([
                    'provider_id' => ['Suspended radiology providers cannot manage scans.'],
                ]);
            }

            if ($scan->branch_id) {
                $branchProviderId = ProviderBranch::query()->whereKey($scan->branch_id)->value('provider_id');

                if ((int) $branchProviderId !== (int) $scan->provider_id) {
                    throw ValidationException::withMessages([
                        'branch_id' => ['The selected branch does not belong to this radiology provider.'],
                    ]);
                }
            }
        });
    }

    public function scopePubliclyVisible(Builder $query): Builder
    {
        return $query
            ->where('radiology_scans.is_active', true)
            ->whereHas('category', fn (Builder $categoryQuery) => $categoryQuery->where('is_active', true))
            ->whereHas('provider', fn (Builder $providerQuery) => $providerQuery
                ->publiclyVisible()
                ->where('type', ProviderType::Radiology));
    }

    public function provider(): BelongsTo
    {
        return $this->belongsTo(Provider::class);
    }

    public function branch(): BelongsTo
    {
        return $this->belongsTo(ProviderBranch::class);
    }

    public function category(): BelongsTo
    {
        return $this->belongsTo(RadiologyScanCategory::class, 'radiology_scan_category_id');
    }

    public function instructions(): HasMany
    {
        return $this->hasMany(RadiologyPreparationInstruction::class);
    }

    public function creator(): BelongsTo
    {
        return $this->belongsTo(User::class, 'created_by');
    }

    public function updater(): BelongsTo
    {
        return $this->belongsTo(User::class, 'updated_by');
    }
}

File: etamen-flutter/lib/features/radiology/domain/entities/radiology_scan.dart
import 'package:etamen_app/features/radiology/domain/entities/radiology_provider_summary.dart';
import 'package:etamen_app/features/radiology/domain/entities/radiology_scan_category.dart';

class RadiologyScan {
  const RadiologyScan({
    required this.id,
    required this.providerId,
    required this.categoryId,
    required this.nameAr,
    this.branchId,
    this.nameEn,
    this.descriptionAr,
    this.descriptionEn,
    this.preparationAr,
    this.preparationEn,
    this.durationMinutes,
    this.basePrice,
    this.requiresPreparation = false,
    this.requiresFasting = false,
    this.contrastRequired = false,
    this.homeAvailable = false,
    this.branchAvailable = true,
    this.reportDeliveryEnabled = true,
    this.isActive = true,
    this.category,
    this.provider,
    this.branch,
  });

  final int id;
  final int providerId;
  final int? branchId;
  final int categoryId;
  final String nameAr;
  final String? nameEn;
  final String? descriptionAr;
  final String? descriptionEn;
  final String? preparationAr;
  final String? preparationEn;
  final int? durationMinutes;
  final String? basePrice;
  final bool requiresPreparation;
  final bool requiresFasting;
  final bool contrastRequired;
  final bool homeAvailable;
  final bool branchAvailable;
  final bool reportDeliveryEnabled;
  final bool isActive;
  final RadiologyScanCategory? category;
  final RadiologyProviderSummary? provider;
  final RadiologyBranchSummary? branch;

  String name(bool isArabic) {
    if (!isArabic && nameEn?.trim().isNotEmpty == true) return nameEn!.trim();
    return nameAr;
  }

  String? description(bool isArabic) {
    if (!isArabic && descriptionEn?.trim().isNotEmpty == true) {
      return descriptionEn!.trim();
    }
    return descriptionAr;
  }

  String? preparation(bool isArabic) {
    if (!isArabic && preparationEn?.trim().isNotEmpty == true) {
      return preparationEn!.trim();
    }
    return preparationAr;
  }

  double get priceValue => double.tryParse(basePrice ?? '') ?? 0;

  String get priceLabel {
    if (basePrice == null || basePrice!.trim().isEmpty) return '-';
    return '$basePrice EGP';
  }
}
